test/protocol/integration/amm/ArrakisUniswapV3AmmAdapter.spec.ts

```diff
@@ -0,0 +1,355 @@
+import "module-alias/register";
+import { BigNumber } from "ethers";
+import { ether } from "@utils/index";
+import { Account } from "@utils/test/types";
+import { Address } from "@utils/types";
+import {
+  ZERO,
+} from "@utils/constants";
+import { AmmModule, ArrakisUniswapV3AmmAdapter } from "@utils/contracts";
+import DeployHelper from "@utils/deploys";
+import {
+  addSnapshotBeforeRestoreAfterEach,
+  getAccounts,
+  getSystemFixture,
+  getUniswapV3Fixture,
+  getArrakisV1Fixture,
+  getWaffleExpect
+} from "@utils/test/index";
+
+import { SystemFixture, UniswapV3Fixture, ArrakisV1Fixture } from "@utils/fixtures";
+
+const expect = getWaffleExpect();
+
+describe("ArrakisUniswapV3AmmAdapter", () => {
```

Comment:
These unit test will need 100% test coverage, have you checked that it covers 100% of the smart contract?

Reply:
It doesn't cover 100% yet. I am still adding more tests.